### PR TITLE
[SYCL-MLIR] Revert move alloca to the beginning of function

### PR DESCRIPTION
**Pull request by whitneywhtsang**
In https://github.com/intel/llvm/pull/8636 and partially https://github.com/intel/llvm/pull/8482, code are added to move alloca to the beginning of function. This PR reverts them, as the original problem (`Basic/linear-sub_group.cpp` failure) is resolved in the SPIRV translator https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1962.

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
There are still E2E test cases that require alloca to be at the beginning of function.
```
Failed Tests (10):
  SYCL :: Assert/assert_in_kernels.cpp
  SYCL :: Assert/assert_in_multiple_tus.cpp
  SYCL :: Assert/assert_in_multiple_tus_one_ndebug.cpp
  SYCL :: Assert/assert_in_one_kernel.cpp
  SYCL :: Assert/assert_in_simultaneous_kernels.cpp
  SYCL :: Assert/assert_in_simultaneously_multiple_tus.cpp
  SYCL :: Assert/assert_in_simultaneously_multiple_tus_one_ndebug.cpp
  SYCL :: ESIMD/assert.cpp
  SYCL :: Graph/Explicit/buffer_ordering.cpp
  SYCL :: Graph/RecordReplay/buffer_ordering.cpp
```

File: polygeist/tools/cgeist/Test/CMakeLists.txt [hunk 1]
```diff
@@ -15,7 +15,6 @@ list(APPEND MLIR_CLANG_TEST_DEPS
   llvm-config 
   llvm-dis
   llvm-as
-  mlir-translate
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Test/CMakeLists.txt:18]: We still need that for host-raising tests

Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Test/CMakeLists.txt:18]: which tests? I tried to grep `mlir-translate` in the `polygeist` folder, but found nothing.

Comment (victor-eds) [reply polygeist/tools/cgeist/Test/CMakeLists.txt:18]: True, then that is enough :+1:

**Review by victor-eds — APPROVED** [review]

**Review by etiotto — APPROVED** [review]

**Pull request merged**